components/director/internal/selfregmanager/selfreg_manager.go

```diff
@@ -85,12 +86,26 @@ func (s *selfRegisterManager) PrepareForSelfRegistration(ctx context.Context, re
 	}
 
 	if _, err := tenant.LoadFromContext(ctx); err == nil && consumerInfo.Flow.IsCertFlow() {
-		distinguishLabel, exists := labels[s.cfg.SelfRegisterDistinguishLabelKey]
-		if !exists {
+		if resourceType == resource.ApplicationTemplate {
+			labels[scenarioassignment.SubaccountIDKey] = consumerInfo.ConsumerID
+		}
+
+		distinguishLabel, distinguishLabelExists := labels[s.cfg.SelfRegisterDistinguishLabelKey]
+		_, productLabelExists := labels[s.appTemplateProductLabel]
+
+		switch {
+		case !distinguishLabelExists && !productLabelExists:
 			if resourceType == resource.Runtime {
 				return labels, nil
```

Comment:
If we are runtime do we want to have `Subaccount` label returned?

Reply:
I do not understand that. On L89 we set the subaccount only if we are in AppTemplate flow. That was the previous behavior as well: 
<img width="1008" alt="image" src="https://user-images.githubusercontent.com/17293744/229109095-2043e9e1-9ef1-4185-8995-dcd30f390056.png">